.github/workflows/test.yml

```
@@ -47,3 +47,34 @@ jobs:
 
     - name: Test Examples
       run: go test -timeout 45s -v -run '^TestExamples$' github.com/go-git/go-git/v6/_examples --examples
+
+  test-wasi:
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.24.x, 1.25.x]
+
+    permissions:
+      contents: read
+
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+    - name: Install Go
+      uses: actions/setup-go@44694675825211faa026b3c33043df3e48a5fa00 # v6.0.0
+      with:
+        go-version: ${{ matrix.go-version }}
+
+    - name: Install Wasmtime
+      run: |
+        curl https://wasmtime.dev/install.sh -sSf | bash
```

Perhaps using their official GHA will ensure the binary lands in the `PATH`:

```suggestion
    - name: Install Wasmtime
      uses: bytecodealliance/actions/wasmtime/setup@v1
```